### ISSUE TITLE
Team vs team setup 

### ISSUE THREAD

**Issue by invicta-nz** [makupi/pavlov-bot#8]
Goal is total setup of SND match by pushing two pre-defined teams to requested sides and then ResetSND. I have POC this with my own simple batchfile process. First define teams in .json file(s). Team will be list of IDs that define all possible players on a team. Command ";matchsetup team1 team2" would look to see if both team names have been defined and if so, loop through members running rcon SwitchTeam to push first team to T-side and second team to CT-side and then after a 5sec pause issue ResetSnd rcon command. 
There may be some common methods between this and currently requested ;batch command as both issue a series of commands. BTW I found with my simple batch processor that I needed to add a short pause before issuing each command. Only .05sec was enough for string of commands, but without it I only got the first one completed and rest ignored. This may not be an issue if you are looping through issuing commands one after another where as I was pushing a pipeline of text.

**Comment by invicta-nz** [comment]
Note that teams often consist of > 5 players while only 5 players generally will be online for a match. So for this reason and because players may randomly already be on the correct side, the expectation is that you will get back lots of "False" replies to the SwitchTeam commands. Best we can do is issue the commands and hope for the best. Ringers (non team stand-in players) are also problematic. The pause before resetSND is for them to check and sort themselves on the right team.

**Comment by invicta-nz** [comment]
This command should be available to Captains and higher

**Comment by invicta-nz** [comment]
It may overly complicate the parsing, but wonder if ringers could be handled with something like ;matchsetup team1 (team2, invicta) where the parens (or some other grouping technique) would add any extra players into the operations for team2

**Comment by makupi** [comment]
Notes from discord convo:    
- Change ;matchsetup help to `;matchsetup <CT teamname> <T teamname> <server name>`
- Change Team members to Steamplayers (to automatically have alias and unique-id
- Command to add temporary members `;ringers add <id/alias> <team>`
- Command to reset temporary members `;ringers reset <team>`
- Command to list teaminfo `;teaminfo <team>`

**Comment by invicta-nz** [comment]
Yeah, maybe should stop reserving ;teams for future team setup and just call that ;teamsetup. So ;teaminfo could become the simpler ;teams and ;teams <teamname>  gives you the playerlist and the team list without args

**Comment by makupi** [comment]
Regarding the permissions for `;ringers add/reset` being "captain" and above. I've added the new "super-roles" `Super-Captain` and `Super-Mod` (names can be changed) that can do all captain and mod commands on all servers. For `ringers add/reset` we don't have a server-name as an argument so I'm only checking for the super-roles now - not sure if that's the best way?     
Additionally I could iterate through all the server names as well and check if they have any `Captain-name` role or anything like that.

**Comment by invicta-nz** [comment]
Yeah, think we need anyone with captain role to have this permission for now, so will need to check for each server. Thinking more generally this is first time we have a function that is less server based and more entire bot based. There may be more if bot becomes self-configuring (although these are likely to be Admin limited).

**Comment by invicta-nz** [comment]
For the super roles, lets keep the current {role}-{context} pattern and call them Captain-bot and Mod-bot

**Comment by makupi** [comment]
Sounds good. Will do those two things tomorrow 
- super roles to `Captain-bot` and `Mod-bot`
- `ringers add/reset` check permissions accepts all server Captains

**Comment by invicta-nz** [comment]
Testing with a "everybody" user with no roles. Attempted to kick and got error: 
```2020-06-25 06:11:40,503 root         INFO     USER: bot-tester#8486 <725590924473663538> -- INVOKED KICK       args: ['1234', 'snd1']
Ignoring exception in on_command_error
Traceback (most recent call last):
  File "/home/steam/.local/share/virtualenvs/pavlov-bot-testing-CtuPEwCD/lib/python3.8/site-packages/discord/ext/commands/core.py", line 83, in wrapped
    ret = await coro(*args, **kwargs)
  File "/home/steam/pavlov-bot-testing/bot/cogs/pavlov.py", line 547, in kick
    if not await check_perm_moderator(ctx, server_name):
  File "/home/steam/pavlov-bot-testing/bot/cogs/pavlov.py", line 111, in check_perm_moderator
    if await check_perm_admin(
  File "/home/steam/pavlov-bot-testing/bot/cogs/pavlov.py", line 67, in check_perm_admin
    for server in _servers.values():
AttributeError: 'list' object has no attribute 'values'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/steam/.local/share/virtualenvs/pavlov-bot-testing-CtuPEwCD/lib/python3.8/site-packages/discord/client.py", line 312, in _run_event
    await coro(*args, **kwargs)
  File "/home/steam/pavlov-bot-testing/bot/__init__.py", line 71, in on_command_error
    raise error
  File "/home/steam/.local/share/virtualenvs/pavlov-bot-testing-CtuPEwCD/lib/python3.8/site-packages/discord/ext/commands/bot.py", line 892, in invoke
    await ctx.command.invoke(ctx)
  File "/home/steam/.local/share/virtualenvs/pavlov-bot-testing-CtuPEwCD/lib/python3.8/site-packages/discord/ext/commands/core.py", line 797, in invoke
    await injected(*ctx.args, **ctx.kwargs)
  File "/home/steam/.local/share/virtualenvs/pavlov-bot-testing-CtuPEwCD/lib/python3.8/site-packages/discord/ext/commands/core.py", line 92, in wrapped
    raise CommandInvokeError(exc) from exc
discord.ext.commands.errors.CommandInvokeError: Command raised an exception: AttributeError: 'list' object has no attribute 'values'
```
Looks to be the case for all commands requiring Captain or higher **except** ringers which rejects with message

**Comment by invicta-nz** [comment]
;anyoneplaying breaks when loading info from this incompletely populated steam page:
https://steamcommunity.com/sharedfiles/filedetails/?id=2111731598


```Jun 25 09:47:41 Pavlov-syd pipenv[377]: 2020-06-25 09:47:41,809 root         INFO     9101 - RCON _recv data='{\n\t"ServerInfo":\n\t{\n\t\t"MapLabel": "UGC2111731598",\n\t\t"GameMode": "SND",\n\t\t"ServerName": "[ArcticVR][NA-WEST]Summer SND1",\n\t\t"RoundState": "Started",\n\t\t"PlayerCount": "0/10"\n\t}\n}\r\n'
Jun 25 09:47:42 Pavlov-syd pipenv[377]: 2020-06-25 09:47:42,721 root         ERROR    Getting map label failed with list index out of range

```

**Comment by makupi** [comment]
Closed through #42
